### ISSUE TITLE
RAP&HRRR tests are not 12 hours

### ISSUE THREAD

**Issue by SamuelTrahanNOAA** [ufs-community/ufs-weather-model#1290]
## Description
A previous PR was intended to reduce RAP and HRRR regression test forecast lengths to 12 hours, but it did not. This is easy to fix, and a PR will appear shortly.

### To Reproduce:
1. Run the tests.
2. Witness 24 hour forecast lengths

**Comment by zach1221** [comment]
Hi, @SamuelTrahanNOAA . Do you if a PR was ever created to address this bug?

**Comment by SamuelTrahanNOAA** [comment]
I fixed the HRRR recently, but it appears three RAP tests are still wrong.

```
grep FHMAX= tests/tests/*rap*|grep 24

tests/tests/rap_2threads_dyn32_phy32:export FHMAX=24
tests/tests/rap_control_dyn32_phy32:export FHMAX=24
tests/tests/rap_restart_dyn32_phy32:export FHMAX=24
```

**Comment by zach1221** [comment]
> I fixed the HRRR recently, but it appears three RAP tests are still wrong.
> 
> ```
> grep FHMAX= tests/tests/*rap*|grep 24
> 
> tests/tests/rap_2threads_dyn32_phy32:export FHMAX=24
> tests/tests/rap_control_dyn32_phy32:export FHMAX=24
> tests/tests/rap_restart_dyn32_phy32:export FHMAX=24
> ```

Ok, I see. If it's just changing the FHMAX export to "12" in the tests/tests/ file , I can make that change in one of the upcoming PRs, so we can resolve this. Do you see any issue with this?

**Comment by SamuelTrahanNOAA** [comment]
> Ok, I see. If it's just changing the FHMAX export to "12" in the tests/tests/ file , I can make that change in one of the upcoming PRs, so we can resolve this. Do you see any issue with this?

You must ensure the results don't change.

**Comment by zach1221** [comment]
> > Ok, I see. If it's just changing the FHMAX export to "12" in the tests/tests/ file , I can make that change in one of the upcoming PRs, so we can resolve this. Do you see any issue with this?
> 
> You must ensure the results don't change.

Of course, I'm testing now on Hera against the existing baselines. I can let you know the results.

**Comment by zach1221** [comment]
Yes, it looks like the results will change. The RESTART files change: /scratch1/NCEPDEV/nems/Zachary.Shrader/ufs-weather-model/tests/logs/RegressionTests_hera.log  
versus the existing: scratch1/NCEPDEV/nems/Zachary.Shrader/ufs-weather-model/tests/logs/RegressionTests_hera.log_existing_baselines

**Comment by gspetro-NOAA** [comment]
@SamuelTrahanNOAA @zach1221 Should this issue be closed? Or might there be a way to change the RAP tests to 12 hours without a baseline change?

**Comment by gspetro-NOAA** [comment]
@SamuelTrahanNOAA The baselines will necessarily change because the restart files will no longer be for the 24th forecast hour but for the 12th. But it looks like the _results_ do not change (e.g., for the `atmf*` and `sfcf*` files). 
However, I think there are other tests that are set to `FHMAX=24` because that is the default value in the `default_vars.sh fv3_v16()` function, which many of the RAP tests inherit. Should the default value be changed in, e.g, `export_rap_common()` so that *all* tests that don't explicitly set a different value have a 12-hour FHMAX? Or was there something particular about these tests that meant they should be set to `FHMAX=12`?

The following RAP tests inherit an FHMAX of 24: 
- rap_2threads
- rap_control
- rap_decomp
- rap_restart
- rap_sfcdiff
- rap_sfcdiff_decomp
- rap_sfcdiff_restart

The three `rap_sfcdiff*` tests look like they should perhaps be 12-hour tests based on `OUTPUT_FH='0 09 12'`. I could explicitly set `FHMAX=12` for those tests if that is desirable. The others all have `OUTPUT_FH='0 21 24'` or the default, so it seems like they are intended to be 24-hour tests.

**Comment by SamuelTrahanNOAA** [comment]
When they were first created, they were all intended to be 12 hour tests. Extending them to 24 was possible, but not necessary. My plan was to change all of them to 12 hour tests in a PR long ago. Somehow I reverted part of those changes, so the tests remain 24 hours.

**Comment by gspetro-NOAA** [comment]
@SamuelTrahanNOAA Ok, then I will change them all to 12-hour tests via a change to the `export_rap()` function. I can make sure results don't change for the ones that already have files for `OUTPUT_FH='0 9 12'`, but for the others, it seems that there would necessarily be a change in results just because the proper baseline files don't exist yet. Do you have a suggestion for how to ensure that the changes are reasonable? 

@MatthewPyle-NOAA Jong suggested that I reach out to you about the RAP cases, too, just to be sure this change is okay from the NCO perspective.

**Comment by MatthewPyle-NOAA** [comment]
@gspetro-NOAA No concerns here about shortening the length of the tests.

**Comment by SamuelTrahanNOAA** [comment]
>  Do you have a suggestion for how to ensure that the changes are reasonable?

The output at 12 hours should be identical regardless of the forecast length.

**Comment by gspetro-NOAA** [comment]
@SamuelTrahanNOAA I understand that, but for a few of the tests, there is no output at 12 hours. For example, the baseline for `rap_control_intel` only has the following files: 
```
atmf000.nc   atmf024.nc	    GFSFLX.GrbF21  GFSPRS.GrbF00  GFSPRS.GrbF24	sfcf000.nc  sfcf024.nc
atmf021.nc   GFSFLX.GrbF00  GFSFLX.GrbF24  GFSPRS.GrbF21  RESTART	sfcf021.nc
```
Nothing for f012.

**Comment by SamuelTrahanNOAA** [comment]
If you want to be thorough, you could add output at hour 12 to the existing 24-hour tests. Then rerun them and you'll have the 12hr output to compare.

**Comment by gspetro-NOAA** [comment]
Great idea! I will do that. :)
